File: common_lib/src/main/java/com/common_lib/base/utils/NetworkUtils.java
package com.common_lib.base.utils;

import android.annotation.TargetApi;
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Build;
import android.provider.Settings;

import com.common_lib.base.BaseConstants;


/**
 * Created by alex on 2016/3/2.
 */
public class NetworkUtils {
    public static final int TYPE_UNKNOWN = -1;

    private static NetworkInfo getActiveNetworkInfo(Context context) {
        if (context == null) {
            return null;
        } else {
            ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            return cm == null ? null : cm.getActiveNetworkInfo();
        }
    }

    private static int getActiveNetworkType(Context context) {
        NetworkInfo info = getActiveNetworkInfo(context);
        return info != null && info.isConnected() ? info.getType() : -1;
    }

    public static boolean isNetworkAvailable(Context context) {
        NetworkInfo info = getActiveNetworkInfo(context);
        return info != null && info.isConnected();
    }

    public static boolean isWiFiConnected(Context context) {
        return getActiveNetworkType(context) == 1;
    }

    @TargetApi(17)
    public static boolean isAirplaneModeOn(Context context) {
        return Build.VERSION.SDK_INT < 17 ? Settings.System.getInt(context.getContentResolver(), "airplane_mode_on", 0) != 0 : Settings.Global.getInt(context.getContentResolver(), "airplane_mode_on", 0) != 0;
    }

    public static boolean checkConnection(Context context) {
        if (isNetworkAvailable(context)) {
            return true;
        } else {
            ToastUtils.showNormalToast(context, BaseConstants.No_Internet);
            return false;
        }
    }
}

File: app/src/main/java/com/novel/read/utlis/NetworkUtils.kt
package com.novel.read.utlis

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

import com.novel.read.base.MyApp

object NetworkUtils {

    /**
     * 获取活动网络信息
     * @return NetworkInfo
     */
    private val networkInfo: NetworkInfo?
        get() {
            val cm = MyApp.context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            return cm.activeNetworkInfo
        }

    /**
     * 网络是否可用
     * @return
     */
    val isAvailable: Boolean
        get() {
            val info = networkInfo
            return info != null && info.isAvailable
        }

    /**
     * 网络是否连接
     * @return
     */
    val isConnected: Boolean
        get() {
            val info = networkInfo
            return info != null && info.isConnected
        }

    /**
     * 判断wifi是否连接状态
     *
     * 需添加权限 `<uses-permission android:name="android.permission
     * .ACCESS_NETWORK_STATE"/>`
     *
     * @param context 上下文
     * @return `true`: 连接<br></br>`false`: 未连接
     */
    fun isWifiConnected(context: Context): Boolean {
        val cm = context
            .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return (cm.activeNetworkInfo != null
                && cm.activeNetworkInfo.type == ConnectivityManager.TYPE_WIFI)
    }


}
